packages/xml-views-validation/src/validators/elements/cardinality-of-aggregation.ts

```diff
@@ -0,0 +1,113 @@
+import { filter, sortBy, map, isEmpty, flatMap } from "lodash";
+import { XMLElement, XMLToken, SourcePosition } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { getUI5AggregationByXMLElement } from "@ui5-language-assistant/logic-utils";
+import { InvalidAggregationCardinalityIssue } from "../../../api";
+import {
+  getMessage,
+  INVALID_AGGREGATION_CARDINALITY,
+} from "../../utils/messages";
+
+export function validateExplicitAggregationCardinality(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): InvalidAggregationCardinalityIssue[] {
+  const aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  // Is xmlElement an aggregation?
+  if (aggregation === undefined || xmlElement.syntax.openName === undefined) {
+    return [];
+  }
+
+  const allSubElementsAtSameDepth = getAllSubElementsAtSameDepth(xmlElement);
+  if (
+    allSubElementsAtSameDepth.length <= 1 ||
+    aggregation.cardinality === "0..n"
+  ) {
+    return [];
+  }
+
+  const firstSubElementAtSameDepth = getFirstSubElementAtSameDepth(
+    allSubElementsAtSameDepth
+  );
+  // The error will not be shown on the first sub element at same depth
+  const redundantAggregationSubElements = filter(
+    xmlElement.subElements,
+    (_) => !isFirstSubElementAtSameDepth(_, firstSubElementAtSameDepth)
+  );
+
+  const invalidAggregationCardinalityIssues = map(
+    redundantAggregationSubElements,
+    (_): InvalidAggregationCardinalityIssue => ({
+      kind: "InvalidAggregationCardinality",
+      message: getMessage(INVALID_AGGREGATION_CARDINALITY, aggregation.name),
+      severity: "error",
+      offsetRange: {
+        start: getSubElementPosition(_).startOffset,
+        end: getSubElementPosition(_).endOffset,
+      },
+    })
+  );
+
+  return invalidAggregationCardinalityIssues;
+}
+
+function getFirstSubElementAtSameDepth(
+  allSubElementsAtSameDepth: XMLElement[]
+): XMLElement {
+  const sortedSubElementsAtSameDepth = sortBy(allSubElementsAtSameDepth, (_) =>
+    getSubElementPosition(_)
+  );
+
+  return sortedSubElementsAtSameDepth[0];
+}
+
+function isFirstSubElementAtSameDepth(
+  subElement: XMLElement,
+  firstSubElement: XMLElement
+): boolean {
+  return (
+    getSubElementPosition(subElement).startOffset ===
+    getSubElementPosition(firstSubElement).startOffset
+  );
+}
+
+function getSubElementPosition(
+  xmlElement: XMLElement
+): XMLToken | SourcePosition {
+  return xmlElement.syntax.openName ?? xmlElement.position;
+}
+
+/**
+ *  @param aggregationElem The aggregation from which we will get all its subElements at the same depth.
+ *
+ * For example - for the following xml snippet the function will return an array of `Toolbar` elemnt and `tnt:ToolHeader` element
```

Comment:
also specify what would be the input for the `following snippet`

> Given XMLSnippet `XYZ` and aggregationElem param `ABC`. `Foo` will be returned 

You can use simpler tag names to make the example easier, e.g:
Uncle1, uncle2, cousin1 consin2